File: backend/src/main/java/com/apurv/request/dto/RequestItemResponse.java
package com.apurv.request.dto;

import java.time.Instant;
import java.util.Map;
import java.util.UUID;

import com.apurv.request.entity.HttpMethod;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class RequestItemResponse {

    private UUID id;
    private String name;
    private HttpMethod method;
    private String url;
    private Map<String, String> headers;
    private String body;
    private UUID collectionId;
    private UUID folderId;
    private UUID ownerId;
    private Instant createdAt;
    private Instant updatedAt;
}

File: frontend/src/types/request/RequestItemResponse.ts
import type { RequestItem } from "./RequestItem";

export interface RequestItemResponse {
  id: string;
  name: string;
  method: RequestItem["method"];
  url: string;
  headers: Record<string, string> | null;
  body: string | null;
  authType: "none" | "bearer" | "basic" | null;
  authValue: Record<string, string> | null;
  timeoutMs: number | null;
  collectionId: string;
  folderId: string | null;
  ownerId: string;
  createdAt: string;
  updatedAt: string;
}